File: go/space-age/space_age.go
// Package space contains utilities for calculating relative orbital times
package space

const earthYearsPerMercuryYear = 0.2408467
const earthYearsPerVenusYear = 0.61519726
const earthYearsPerMarsYear = 1.8808158
const earthYearsPerJupiterYear = 11.862615
const earthYearsPerSaturnYear = 29.447498
const earthYearsPerUranusYear = 84.016846
const earthYearsPerNeptuneYear = 164.79132

const secondsPerEarthYear = 31557600

var planets = map[Planet]float64{
	"Mercury": secondsPerEarthYear * earthYearsPerMercuryYear,
	"Venus":   secondsPerEarthYear * earthYearsPerVenusYear,
	"Earth":   secondsPerEarthYear,
	"Mars":    secondsPerEarthYear * earthYearsPerMarsYear,
	"Jupiter": secondsPerEarthYear * earthYearsPerJupiterYear,
	"Saturn":  secondsPerEarthYear * earthYearsPerSaturnYear,
	"Uranus":  secondsPerEarthYear * earthYearsPerUranusYear,
	"Neptune": secondsPerEarthYear * earthYearsPerNeptuneYear,
}

// Age calculates the number of years that have passed on the provided Planet for the given number of seconds
func Age(seconds float64, planet Planet) float64 {
	secondsPerPlanetsYear, found := planets[planet]
	if found {
		return seconds / secondsPerPlanetsYear
	}
	return 0
}

type Planet string

File: java/space-age/src/main/java/SpaceAge.java
public class SpaceAge {
    private static double EARTH_SECONDS_PER_YEAR = 31557600;
    private enum Planet {
        MERCURY(0.2408467),
        VENUS(0.61519726),
        EARTH(1.0),
        MARS(1.8808158),
        JUPITER(11.862615),
        SATURN(29.447498),
        URANUS(84.016846),
        NEPTUNE(164.79132);

        private final double secondsPerYear;
        Planet(double orbitalPeriod) {
            secondsPerYear = orbitalPeriod * EARTH_SECONDS_PER_YEAR;
        }
        public double getSecondsPerYear() {
            return secondsPerYear;
        }
    }
    private double seconds;

    public SpaceAge(double seconds) {

        this.seconds = seconds;
    }

    public double getSeconds() {
        return seconds;
    }

    public double onEarth() {
        return yearsOn(Planet.EARTH);
    }

    public double onMercury() {
        return yearsOn(Planet.MERCURY);
    }

    public double onVenus() {
        return yearsOn(Planet.VENUS);
    }

    public double onMars() {
        return yearsOn(Planet.MARS);
    }

    public double onJupiter() {
        return yearsOn(Planet.JUPITER);
    }

    public double onSaturn() {
        return yearsOn(Planet.SATURN);
    }

    public double onUranus() {
        return yearsOn(Planet.URANUS);
    }

    public double onNeptune() {
        return yearsOn(Planet.NEPTUNE);
    }

    private double yearsOn(Planet planet) {
        return seconds / planet.getSecondsPerYear();
    }
}
